Sources/Kitura/RouterElement.swift

```diff
@@ -124,7 +134,8 @@ class RouterElement {
     /// - Parameter match: the regular expression result
     /// - Parameter request:
     private func setParameters(forRequest request: RouterRequest, fromUrlPath urlPath: NSString, match: TextChekingResultType) {
-        var parameters = [String:String]()
+        var parameters = mergeParams ? request.parameters : [:]
```

Comment:
Once in a router that is having the parameters merged, each and every subsequent RouteHAndler or middleware will get the cumulative set of the parameters from the top level router and every path that matched. The original design was that each RouteHandler or middleware gets its own parameters and no-one else's. I understand that here we adding in the parameters from the parent route, but are we adding in all of the previous routes as well.
Some examples might help.
```swift
  let router = Router()
  router.get("/path/:p1/:p2", handler: handler1)
  router.get("/path/:p1/testing", handler: handler2)
``` 
In this case, for the same URL path, both handler1 and handler2 will be invoked. When handler1 is invoked there will be two elements in request.parameters, p1 and p2. When handler2 is invoked there will be one element in requests.parameters, p1.

```swift
  let router = Router()
  let subRouter = router.route("/path/:p1", mergeParameters: true)
  subRouter.get("/:p2", handler: handler1)
  subRouter.get("/testing", handler: handler2)
``` 
In this case, for the same URL path, both handler1 and handler2 will be invoked. When handler1 is invoked there will be two elements in request.parameters, p1 and p2. But now when handler2 is invoked there will **also** be **two* elements in requests.parameters, p1 and p2, **instead of** just p1.